mirrord/cli/src/db_branches.rs

```diff
@@ -0,0 +1,236 @@
+use std::collections::HashSet;
+
+use kube::{
+    Api,
+    api::{DeleteParams, ListParams},
+    client::ClientBuilder,
+};
+use mirrord_config::{LayerConfig, config::ConfigContext};
+use mirrord_kube::{api::kubernetes::create_kube_config, retry::RetryKube};
+use mirrord_operator::crd::mysql_branching::MysqlBranchDatabase;
+use mirrord_progress::{Progress, ProgressTracker};
+use prettytable::{Table, row};
+use tower::{buffer::BufferLayer, retry::RetryLayer};
+
+use crate::{
+    CliError, CliResult,
+    config::{DbBranchesArgs, DbBranchesCommand},
+};
+
+pub async fn db_branches_command(args: DbBranchesArgs) -> CliResult<()> {
+    match &args.command {
+        DbBranchesCommand::Status { names } => status_command(&args, names.as_slice()).await,
+        DbBranchesCommand::Destroy { all, names } => destroy_command(&args, *all, names).await,
+    }
+}
+
+async fn status_command(args: &DbBranchesArgs, names: &[String]) -> CliResult<()> {
+    let names: HashSet<_> = names.iter().collect();
+
+    let mut progress = ProgressTracker::from_env("DB Branches Status");
+    let mut status_progress = progress.subtask("fetching branches");
+
+    let mut cfg_context = ConfigContext::default()
+        .override_env_opt(LayerConfig::FILE_PATH_ENV, args.config_file.clone())
+        .override_env_opt("MIRRORD_TARGET_NAMESPACE", args.namespace.clone());
+
+    let mut layer_config = LayerConfig::resolve(&mut cfg_context)?;
+    layer_config.agent.privileged = true;
```

Comment:
I think this one, `layer_config.agent.privileged`,  is VPN command specific. We just need to resolve the layer config and build a client with it.